dbos/_core.py

```diff
@@ -1156,27 +1156,15 @@ def check_existing_result() -> Union[NoResult, R]:
 
         @wraps(func)
         def wrapper(*args: Any, **kwargs: Any) -> Any:
-            rr: Optional[str] = check_required_roles(func, fi)
-            # Entering step is allowed:
-            #  No DBOS, just call the original function directly
-            #  In a step already, just call the original function directly.
-            #  In a workflow (that is not in a step already)
-            #  Not in a workflow (we will start the single op workflow)
-            if not dbosreg.dbos or not dbosreg.dbos._launched:
-                # Call the original function directly
-                return func(*args, **kwargs)
+            # If the step is called from a workflow, run it as a step.
+            # Otherwise, run it as a normal function.
             ctx = get_local_dbos_context()
-            if ctx and ctx.is_step():
-                # Call the original function directly
-                return func(*args, **kwargs)
-            if ctx and ctx.is_within_workflow():
-                assert ctx.is_workflow(), "Steps must be called from within workflows"
+            if ctx and ctx.is_workflow():
+                rr: Optional[str] = check_required_roles(func, fi)
                 with DBOSAssumeRole(rr):
                     return invoke_step(*args, **kwargs)
             else:
-                tempwf = dbosreg.workflow_info_map.get("<temp>." + step_name)
-                assert tempwf
-                return tempwf(*args, **kwargs)
+                return func(*args, **kwargs)
```

Comment:
Do we also want to clean up some registration below?

Reply:
Not doable as far as I know, still need temp workflows for queues